lambda/chaos.py

```diff
@@ -0,0 +1,306 @@
+            status = invk['Status']
+            if status == 'Success':
+                pass
+            elif status in ['Pending', 'InProgress', 'Delayed']:
+                raise Retry(f"Command { cmd_id } is in status: { status }")
+            else:
+                raise Fail(f"Command { cmd_id } failed: { status }")
+    # for FIS run experiment
+    elif 'experiment' in run:
+        exp=run['experiment']
+        exp_id=exp['id']
+        res=fis.get_experiment(id=exp_id)
+        status=res['experiment']['state']['status']
+        if status == 'completed':
+            res={'result': status}
+        else:
+            raise Retry(f"experiment is in status: {status}")
+    else:
+        raise Fail('wait_for_finish: unsupported inputs')
+    return res
+
+
+def to_json(d:dict):
+    return json.dumps(d, default=str)
+
+def dict_event(evt):
+    """
+    Ensure the input of lambda is a dict.
+    """
+    if isinstance(evt , str):
+        return json.loads(evt)
+    if isinstance(evt, dict):
+        return evt
+    else:
+        raise Fail("unsupported input for lambda: %s", type(evt))
+
+##################################
+## Lambda handlers
+##  event and context are dict()
+##  output must be json
+###################################
+def random_fault() -> str:
+    return random.choice(FAULTS)
+
+def handler_start_traffic(event, context):
+    """
+    Event Input:
+       - traffic_args : str
+    Output: Async command call ret for polling
+    """
+    event = dict_event(event)
+    cluster_name=os.environ['cluster_name']
+    traffic_args=event['traffic_args']
+    lb = '.'.join(['lb', 'int', cluster_name])
+    traffic_args['Host']=[lb]
+    return to_json(start_traffic(cluster_name, traffic_args))
+
+def handler_stop_traffic(event, context):
+    """
+    Event Input: -
+    Output: Async Command call ret for polling
+    """
+    event = dict_event(event)
+    cluster_name=os.environ['cluster_name']
+    return to_json(stop_traffic(cluster_name))
+
+def handler_poll_result(event, context):
+    """
+    Poll sync call return value, either normal return
+    or throw Errors for Retry or Fail
+    Event Input:
+       - Payload : dict
+    Output: -
+    """
+    event = dict_event(event)
+    call_res=json.loads(event['Payload'])
+    return to_json(poll_result(call_res))
+
+def handler_check_traffic(event, context):
+    """
+    Query prometheus and check if the traffic is ok
+
+    Event Input:
+       - time : timestamp
+       - period: range of time for looking back
+             see https://prometheus.io/docs/prometheus/latest/querying/examples/
+    Output: -
+    """
+
+    event = dict_event(event)
+    if 'time' in event:
+        time_at=event['time']
+    else:
+        time_at=time.time()
+    period=event['period']
+    cluster_name=os.environ['cluster_name']
+    if 'prom_host' in os.environ:
+        prom_host=os.environ['prom_host']
+    else:
+        prom_host=f"lb.int.{cluster_name}:9090"
```

Comment:
Tiny nit suggestion:

```suggestion
    prom_host=os.environ.get('prom_host', f"lb.int.{cluster_name}:9090")
```